### DISCUSSION TITLE
new bootlogo swapping code

### DISCUSSION THREAD

**Discussion by EasyIP2023** [OE4T/meta-tegra#670] [Q&A] [answered]
Having trouble swapping out the stock logo with the latest code on branch dunfell-l4t-r32.5.0

So in my layer I have a reference to ```PREFERRED_PROVIDER_virtual/bootlogo = "bootlogo-custom"``` in a .conf.

External environment variables are
```
export BB_PRESERVE_ENV=1
export OS_FILES=$(pwd)
```

I've also added a ```recipes-bsp/bootlogo/bootlogo-custom_%.bbappend``` file as seen bellow.
```
# BMP_BLOB_SIZES ?= "1920x1080 1280x720 640x480"
IMAGE_NAME="WallpaperDog-10778246.jpg"

FILESEXTRAPATHS_prepend := "${OS_FILES}/logos:"
SRC_URI += "file://${IMAGE_NAME}"

PACKAGECONFIG = "raw"
RAW_GRAPHIC_FILE = "${IMAGE_NAME}"
SPLASH_GRAPHIC_BACKGROUND = "black"

do_configure() {
  mv ${WORKDIR}/${IMAGE_NAME} ${S}
}
```

This appears to use imagemagick to compress and convert an arbitrary picture to a *.blob file, but when I boot up. I still see the stock logo. Is there something that I am missing or doing wrong? Wondering does it has anything to do with the amount of bytes the image I choose has? Or for say a u-boot issue?

Used a 1920x1080 logo from here: https://wallpaper.dog/gigabyte-logo-wallpapers

**Comment by ichergui** [comment]
Hey @EasyIP2023 
which machine are you using ?
I'm asking just to try to reproduce what you are doing ?

**Reply by EasyIP2023** [reply]
jetson-xavier-nx-devkit

**Comment by kekiefer** [comment, marked as answer]
After building this, how are you installing it? The update payloads don't ship the splash image, but the main flash script will.

Check out that this package is built, it'll be put in the `$DEPLOYDIR`. For testing, you can copy the `bootlogo-${MACHINE}.blob` to the target and dd it to `/dev/disk/by-partlabel/BMP` and `/dev/disk/by-partlabel/BMP_b`

**Reply by EasyIP2023** [reply]
Not doing anything special at the end of builds. Just running ```./dosdcard <block dev>```. I can see the .blob files were generated.

Okay awesome! And thanks! So, ```dosdcard.sh``` just isn't dd'ing to the partitions (BMP/BMP_b) by default? I've actually never investigated that.

```
<path to build dir>/tmp/deploy/images/jetson-xavier-nx-devkit/bootlogo-jetson-xavier-nx-devkit-1.0-r0.blob
<path to build dir>/tmp/deploy/images/jetson-xavier-nx-devkit/bootlogo-jetson-xavier-nx-devkit.blob
```

**Reply by EasyIP2023** [reply]
O I see now after looking at the partition config file for jetson-xavier-nx-devkit. Okay I should be using ```doflash.sh```. cool cool

**Reply by madisongh** [reply]
Yep, on the Xavier NX, the BMP partitions are in the SPI flash.
